Libplanet.Tests/Net/SwarmTest.cs

```diff
@@ -2590,6 +2590,44 @@ public async Task DoNotReceiveBlockFromNodeHavingDifferenceGenesisBlock()
             }
         }
 
+        [Fact(Timeout = Timeout)]
+        public async Task ThrowInvalidGenesisException()
+        {
+            var policy = new BlockPolicy<DumbAction>();
+            BlockChain<DumbAction> MakeBlockChain() => TestUtils.MakeBlockChain(
+                policy,
+                new DefaultStore(path: null),
+                null,
+                new PrivateKey());
+
+            var chainA = MakeBlockChain();
+            var chainB = MakeBlockChain();
+            var swarmA = CreateSwarm(chainA);
+            var swarmB = CreateSwarm(chainB);
+
+            await chainB.MineBlock(_fx1.Address1);
+
+            await StartAsync(swarmA);
+            await StartAsync(swarmB);
+
+            await swarmA.AddPeersAsync(new[] { swarmB.AsPeer }, null);
+            Assert.NotEqual(chainA.Genesis, chainB.Genesis);
+            try
+            {
+                await swarmA.PreloadAsync();
+            }
+            catch (AggregateException exception)
```

Comment:
In this way, the test passes without throwing the exception.

How about this?

```
            Task t = swarmA.PreloadAsync();
            await Assert.ThrowsAsync<AggregateException>(async () => await t);
            var exception = t.Exception.InnerException?.InnerException;
            Assert.IsType<InvalidGenesisBlockException>(exception);
```

Reply:
Right.